File: LeetCode/cpp/1833_Maximum_Ice_Cream_Bars.cpp
class Solution {
public:
	int maxIceCream(vector<int>& costs, int coins) {
		int answer = 0;
		map<int, int> um;

		for (int cost : costs) {
			um[cost]++;
		}

		for (auto m : um) {
			if (coins >= m.first) {
				for (int i = 0; i < m.second; i++) {
					if (coins < m.first) {
						break;
					}

					coins -= m.first;
					answer++;
				}
			}
		}

		return answer;
	}
};

File: LeetCode/java/1833_Maximum_Ice_Cream_Bars.java
import java.util.Map;
import java.util.TreeMap;

class Solution {
    public int maxIceCream(int[] costs, int coins) {
        int answer = 0;
        Map<Integer, Integer> map = new TreeMap<>();

        for (int cost : costs) {
            map.put(cost, map.getOrDefault(cost, 0) + 1);
        }

        for (int key : map.keySet()) {
            if (coins >= key) {
                for (int i = 0; i < map.get(key); i++) {
                    if (coins < key) {
                        break;
                    }

                    coins -= key;
                    answer++;
                }
            }
        }

        return answer;
    }
}